plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py

```diff
@@ -209,20 +242,42 @@ def eval_deisolate(self, port_name):
         self.ports_states.pop(port_name)
         self.logger.warning(f"Deisolated port: {port_name}. dry_run: {self.dry_run}")
                 
-    def get_rate_and_update(self, port_name, counter_name, new_val):
-        port_data = self.ports_data.get(port_name)
-        if port_data:
-            old_val = port_data.get(counter_name)
+    def get_rate_and_update(self, port_name, counter_name, new_val, timestamp):
+        port_obj = self.ports_data.get(port_name)
+        if port_obj:
+            if timestamp - port_obj.last_timestamp < 1:
+                return port_obj.counters_values.get(counter_name + "_rate", 0)
+            old_val = port_obj.counters_values.get(counter_name)
             if old_val and new_val > old_val:
-                counter_delta = (new_val - old_val) / self.t_isolate
+                counter_delta = (new_val - old_val) / (timestamp - port_obj.last_timestamp)
             else:
                 counter_delta = 0
         else:
             self.ports_data[port_name] = {}
             counter_delta = 0
-        self.ports_data.get(port_name)[counter_name] = new_val
+            port_obj = self.ports_data[port_name]
+        port_obj.counters_values[counter_name] = new_val
+        port_obj.counters_values[counter_name + "_rate"] = counter_delta
+        port_obj.counters_values[Constants.LAST_TIMESTAMP] = timestamp
         return counter_delta
-
+   
+    def check_link_down_condition(self, port_obj, port_name, ports_counters):
+        peer_port_name = port_obj.peer
+        if not peer_port_name or peer_port_name == 'N/A':
+            return None
```

Comment:
this check should be done before
  peer_port_name = port_obj.peer

Reply:
done